pallets/sponsorship/src/lib.rs

```diff
@@ -349,7 +375,14 @@ pub mod pallet {
 					user_details.reserve_quota.balance(),
 				)?;
 				pot_details.reserve_quota.saturating_sub(repaid);
-				Self::remove_user(&pot, &user);
+				UserRegistrationCount::<T>::mutate(&user, |count| {
+					count.saturating_dec();
+					if count.is_zero() {
```

Comment:
if count was zero, and this was called in an unexpected way this will be called twice.

Reply:
It doesn't happen as count must be at least 1 at this point. I will however make a slight change for that unlikely case.